qiskit/pulse/transforms/alignments.py

```diff
@@ -86,21 +100,31 @@ def align(self, schedule: Schedule) -> Schedule:
             self._push_left_append(aligned, child)
         return aligned
 
-    @staticmethod
-    def _push_left_append(this: Schedule, other: ScheduleComponent) -> Schedule:
+    def _push_left_append(self, this: Schedule, other: ScheduleComponent) -> Schedule:
         """Return ``this`` with ``other`` inserted at the maximum time over
         all channels shared between ```this`` and ``other``.
 
         Args:
             this: Input schedule to which ``other`` will be inserted.
             other: Other schedule to insert.
+            ignore_frames: If true then frame instructions will be ignore. This
+                should be set to true if the played Signals in this context
+                do not share any frames.
 
         Returns:
             Push left appended schedule.
         """
         this_channels = set(this.channels)
         other_channels = set(other.channels)
         shared_channels = list(this_channels & other_channels)
+
+        # Conservatively assume that a Frame instruction could impact all channels
```

Comment:
Yes I think so. This however means that you need to bring in the `frames_config` at schedule construction time. This could be a good future extension.